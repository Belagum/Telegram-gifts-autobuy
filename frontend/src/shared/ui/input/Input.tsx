// SPDX-License-Identifier: Apache-2.0
// Copyright 2025 Vova Orig

import type { InputHTMLAttributes } from "react";
import React from "react";
import clsx from "clsx";
import "./input.css";

export interface InputProps
  extends Omit<InputHTMLAttributes<HTMLInputElement>, "prefix" | "suffix"> {
  error?: string;
  leadingAddon?: React.ReactNode;
  trailingAddon?: React.ReactNode;
}

export const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ className, error, leadingAddon, trailingAddon, ...rest }, ref) => {
    return (
      <div className={clsx("ui-input", { "ui-input--error": Boolean(error) })}>
        {leadingAddon && <span className="ui-input__prefix">{leadingAddon}</span>}
        <input ref={ref} className={clsx("ui-input__field", className)} {...rest} />
        {trailingAddon && <span className="ui-input__suffix">{trailingAddon}</span>}
        {error && <div className="ui-input__error">{error}</div>}
      </div>
    );
  },
);

Input.displayName = "Input";
