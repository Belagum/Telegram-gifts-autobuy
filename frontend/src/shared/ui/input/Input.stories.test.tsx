// SPDX-License-Identifier: Apache-2.0
// Copyright 2025 Vova Orig

import { render } from "@testing-library/react";
import { Input } from "./Input";
import * as stories from "./Input.stories";

describe("Input stories", () => {
  it("renders default input", () => {
    const { container } = render(<Input {...(stories.Default.args ?? {})} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders error state", () => {
    const { container } = render(<Input {...(stories.WithError.args ?? {})} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
