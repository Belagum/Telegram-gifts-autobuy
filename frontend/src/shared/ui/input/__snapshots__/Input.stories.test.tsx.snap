// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input stories > renders default input 1`] = `
<div
  class="ui-input"
>
  <input
    class="ui-input__field"
  />
</div>
`;

exports[`Input stories > renders error state 1`] = `
<div
  class="ui-input ui-input--error"
>
  <input
    class="ui-input__field"
  />
  <div
    class="ui-input__error"
  >
    Ошибка валидации
  </div>
</div>
`;
